src/main/java/org/broadinstitute/consent/http/service/ConsentService.java

```diff
@@ -100,6 +227,47 @@ public void updateConsentDac(String consentId, Integer dacId) {
         return dacService.filterConsentManageByDAC(consentManageList, authUser);
     }
 
+    public List<ConsentAssociation> getAssociation(String consentId, String associationType, String objectId) {
+        logger.trace(String.format("getAssociation consentId='%s' associationType='%s', objectId='%s'",
+                consentId, associationType, objectId));
+
+        checkConsentExists(consentId);
+        List<ConsentAssociation> result;
+        if (associationType == null)
+            result = getAllAssociationsForConsent(consentId);
+        else {
+            result = new ArrayList<>();
+            List<String> id_list;
+            if (objectId == null)
+                id_list = consentDAO.findAssociationsByType(consentId, associationType);
+            else { // both associationType and objectId specified
+                id_list = new ArrayList<>();
+                if (consentDAO.findAssociationByTypeAndId(consentId, associationType, objectId) != null)
+                    id_list.add(objectId);
+            }
+            result.add(new ConsentAssociation(associationType, id_list));
+        }
+        return result;
+    }
+
+    public Consent retrieve(String id) throws UnknownIdentifierException {
+        Consent consent = consentDAO.findConsentById(id);
+        if (consent == null) {
+            throw new UnknownIdentifierException(String.format("Could not find consent with id %s", id));
+        }
+
+        Election election = electionDAO.findLastElectionByReferenceIdAndType(id, ElectionType.TRANSLATE_DUL.getValue());
+        if (election != null) {
+            consent.setLastElectionStatus(election.getStatus());
+            consent.setLastElectionArchived(election.getArchived());
+        }
+        return consent;
+    }
+
+    public Consent getConsentFromDatasetID(Integer datasetId) {
+        return consentDAO.findConsentFromDatasetID(datasetId);
+    }
```

Comment:
Seems like the only usage of this is in `AbstractMatchingServiceAPI`. Can we instead just make the dao call in that class and drop this method completely from the service?